src/modules/navigator/mission_block.cpp

```diff
@@ -469,10 +477,14 @@ MissionBlock::issue_command(const mission_item_s &item)
 }
 
 float
-MissionBlock::get_time_inside(const struct mission_item_s &item)
+MissionBlock::get_time_inside(const mission_item_s &item) const
 {
-	if (item.nav_cmd != NAV_CMD_TAKEOFF) {
-		return item.time_inside;
+	if ((item.nav_cmd == NAV_CMD_WAYPOINT && _navigator->get_vstatus()->is_rotary_wing) ||
+	    item.nav_cmd == NAV_CMD_LOITER_TIME_LIMIT ||
+	    item.nav_cmd == NAV_CMD_DELAY) {
+
+		// TODO: set appropriate upper limit
+		return math::constrain(item.time_inside, 0.0f, 3600.0f);
```

Comment:
Where does this magic number come from?

I'd prefer something like:
```
static constexpr float upper_bound_for_sanity = 3600.0f;
```

Reply:
We don't even necessarily need an upper limit, just seemed like it would almost certainly be garbage past a certain point.

Reply:
If anything it would come back from the spec.

Reply:
Dropped.